packages/asset-swapper/contracts/src/LidoSampler.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+
+  Copyright 2021 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6;
+pragma experimental ABIEncoderV2;
+
+import "./SamplerUtils.sol";
+
+contract LidoSampler is SamplerUtils {
+    struct LidoInfo {
+        address stEthToken;
+        address wethToken;
+    }
+
+    /// @dev Sample sell quotes from Lido
+    /// @param lidoInfo Info regarding a specific Lido deployment
+    /// @param takerToken Address of the taker token (what to sell).
+    /// @param makerToken Address of the maker token (what to buy).
+    /// @param takerTokenAmounts Taker token sell amount for each sample.
+    /// @return makerTokenAmounts Maker amounts bought at each taker token
+    ///         amount.
+    function sampleSellsFromLido(
+        LidoInfo memory lidoInfo,
+        address takerToken,
+        address makerToken,
+        uint256[] memory takerTokenAmounts
+    )
+        public
+        pure
+        returns (uint256[] memory makerTokenAmounts)
+    {
+        _assertValidPair(makerToken, takerToken);
+
+        uint256 numSamples = takerTokenAmounts.length;
+        makerTokenAmounts = new uint256[](numSamples);
+        if (takerToken != lidoInfo.wethToken || makerToken != address(lidoInfo.stEthToken)) {
+            // Return 0 values if not selling WETH for stETH
+            return makerTokenAmounts;
+        }
+
+        for (uint256 i = 0; i < numSamples; i++) {
+            // Minting stETH is always 1:1
+            makerTokenAmounts[i] = takerTokenAmounts[i];
```

Comment:
Oh interesting. Does lido reward LPs purely with eth2 validator rewards?

Reply:
Actually, this is quite complex. When you deposit ETH for stETH you get a certain amount of shares, at the time of deposit this translates to 1:1 ETH to stETH, but as the validators collect more staking rewards your shares will be worth more stETH. The `balanceOf` function dynamically translates shares -> stETH 😅 

For LPing on Curve, you also receive their governance token Lido token.

Reply:
FWIW I think `return takerTokenAmounts;` here would accomplish the same thing, without having to iterate through the for loop.